pontoon/sync/vcs/models.py

```diff
@@ -293,6 +295,33 @@ def resources_for_path(self, path):
                 if is_resource(filename):
                     yield os.path.join(root, filename)
 
+    def create_locale_directory(self, locale, path):
+        if not self.db_project.has_multi_locale_repositories:
+            source_directory = self.source_directory_path()
+            parent_directory = os.path.abspath(os.path.join(source_directory, os.pardir))
+            locale_directory = os.path.join(parent_directory, locale.code.replace('-', '_'))
+
+            # For asymmetric formats, create empty folder
+            if is_asymmetric_resource(path):
+                os.makedirs(locale_directory)
+
+            # For other formats, copy resources from source directory
+            else:
+                shutil.copytree(source_directory, locale_directory)
+
+                for root, dirnames, filenames in os.walk(locale_directory):
+                    for filename in filenames:
+                        path = os.path.join(root, filename)
+                        if is_resource(filename):
+                            os.rename(path, source_to_locale_path(path))
+                        else:
+                            os.remove(path)
+
+            return locale_directory
+
+        else:
+            raise IOError('Directory for locale `{0}` not found'.format(locale.code))
```

Comment:
Maybe It's my thing but i would add separate class for exception, something like a ```

class MissingLocaleDirectory(IOError)
    pass```
Just because it's easier to filter out (e.g. during testing) this specific case from the all other IOErrors that can happen here.